### PR TITLE
Closes several issues regarding releases/movies linking

### LINKED ISSUE #55: Match Discogs releases
Currently, only **12** out of **256k** albums in Wikidata have a Discogs identifier.

Discogs has a huge amount of albums: we could either leverage the `releases_url` tag of each `artist` or the `releases` dump directly, e.g.:
https://discogs-data.s3-us-west-2.amazonaws.com/data/2018/discogs_20180801_releases.xml.gz

### LINKED ISSUE #56: Match MusicBrainz releases
Currently, only **37** out of **256k** albums in Wikidata have a MusicBrainz identifier:
```
SELECT (COUNT(DISTINCT ?item) AS ?count) WHERE { ?item wdt:P31/wdt:P279* wd:Q482994 ; wdt:P434 ?identifier . }
```
Investigate how we could leverage the album data in MusicBrainz to **both** populate album identifiers and _performer_ statements, like:
[Animals](https://www.wikidata.org/wiki/Q206216), [performer](https://www.wikidata.org/wiki/Property:P175), [Pink Floyd](https://www.wikidata.org/wiki/Q2306)

### LINKED ISSUE #80: Album matcher
- [x] Get Wikidata musical albums;
- [ ] implement #55;
- [ ] implement #56;
- [ ] run name-based matchers.

### LINKED ISSUE #282: Match IMDb titles
Investigate how to exploit IMDb titles (movies) data to both populate movie identifiers and relevant statements, e.g., [El día de la bestia](https://www.wikidata.org/wiki/Q1312929) [director](https://www.wikidata.org/wiki/Property:P57) [Álex de la Iglesia](https://www.wikidata.org/wiki/Q250627)

Relevant statements mappings can be built upon occupations, e.g., director -> P57, etc.

### PR DESCRIPTION
**Pull request by MaxFrax**
Closes #55 
Closes #56 
Closes #80 
Closes #282

### CODE REVIEW

File: soweego/commons/constants.py [hunk 1]
```diff
@@ -117,15 +131,25 @@
...
     }
 }
 
 # When building the wikidata dump for catalogs in this array
 # also the QIDs of a person's occupations will be included
 # as part of the dump
-REQUIRE_OCCUPATIONS = [
-    'imdb'
+REQUIRE_OCCUPATIONS = {
+    'imdb': ['actor', 'director', 'musician', 'producer', 'write']
```
Comment (marfox) [inline soweego/commons/constants.py]:
```suggestion
    'imdb': ['actor', 'director', 'musician', 'producer', 'writer']
```

File: soweego/importer/discogs_dump_extractor.py [hunk 2]
```diff
@@ -48,76 +48,145 @@ class DiscogsDumpExtractor(BaseDumpExtractor):
...
+
+            if not node.tag == 'master':
+                continue
+
+            self.total_entities += 1
+            entity = discogs_entity.DiscogsMasterEntity()
+            entity.catalog_id = node.attrib['id']
+            genres = set()
+            for child in node:
+                if child.tag == "main_release":
```
Comment (marfox) [inline soweego/importer/discogs_dump_extractor.py]:
```suggestion
                if child.tag == 'main_release':
```

File: soweego/importer/imdb_dump_extractor.py [hunk 3]
```diff
@@ -118,17 +118,33 @@ def extract_and_populate(self, dump_file_paths: List[str], resolve: bool) -> Non
...
+            movie_entity.name_tokens = ' '.join(text_utils.tokenize(movie_info.get('primaryTitle')))
             movie_entity.is_adult = True if movie_info.get(
                 'isAdult') == '1' else False
-            movie_entity.start_year = movie_info.get('startYear')
-            movie_entity.end_year = movie_info.get('endYear')
+            try:
+                movie_entity.born = datetime.date(year=int(movie_info.get('startYear')), month=1, day=1)
+                movie_entity.born_precision = 9
+            except:
+                pass
```
Comment (marfox) [inline soweego/importer/imdb_dump_extractor.py]:
```suggestion
                LOGGER.debug('No start year value for %s', movie_entity)
                pass
```

File: soweego/importer/imdb_dump_extractor.py [hunk 4]
```diff
@@ -118,17 +118,33 @@ def extract_and_populate(self, dump_file_paths: List[str], resolve: bool) -> Non
...
+            try:
+                movie_entity.end_year = datetime.date(year=int(movie_info.get('endYear')), month=1, day=1)
```
Comment (marfox) [inline soweego/importer/imdb_dump_extractor.py]:
```suggestion
                movie_entity.died = datetime.date(year=int(movie_info.get('endYear')), month=1, day=1)
```
The `end_year` field was removed in the DB entity model, so I assume it wasn't updated here.

File: soweego/importer/imdb_dump_extractor.py [hunk 5]
```diff
@@ -118,17 +118,33 @@ def extract_and_populate(self, dump_file_paths: List[str], resolve: bool) -> Non
...
+                movie_entity.died_precision = 9
+            except:
+                pass
```
Comment (marfox) [inline soweego/importer/imdb_dump_extractor.py]:
```suggestion
                LOGGER.debug('No end year value for %s', movie_entity)
                pass
```

File: soweego/linker/workflow.py [hunk 6]
```diff
@@ -174,9 +174,11 @@ def in_both_datasets(col: str) -> bool:
                                      occupations_col_name,
                                      label='occupation_qids'))
 
-    feature_vectors = compare.compute(
-        candidate_pairs, wikidata, target).drop_duplicates()
-        
+    if in_both_datasets(constants.GENRE):
+        compare.add(SimilarTokens(constants.GENRE,
```
Comment (marfox) [inline soweego/linker/workflow.py]:
```suggestion
        # Feature 9: genre similar tokens
        compare.add(SimilarTokens(constants.GENRE,
```

File: soweego/importer/imdb_dump_extractor.py (hunk 4 above)
Comment (MaxFrax) [reply soweego/importer/imdb_dump_extractor.py]: You're right!

**Pull request merged**